crates/ide/src/completion/presentation.rs

```diff
@@ -191,6 +191,22 @@ impl Completions {
         func: hir::Function,
         local_name: Option<String>,
     ) {
+        fn add_arg(arg: &str, ty: &Type, ctx: &CompletionContext) -> String {
+            let mut prefix = "";
+            if let Some(derefed_ty) = ty.remove_ref() {
+                ctx.scope.process_all_names(&mut |name, scope| {
```

Comment:
Let's lift this logic out of the loop, to not re-do it for each arguments. I think we want to collect all `Local`s to a vec. 

I *think* we might actually do this processing elsewhere already (to compute score), so maybe it makes sense to add `locals` field to `CompletionContext`.

Reply:
Changed it to put locals as a field in `CompletionContext`.